### PR TITLE
Add Akka.Discovery.Config.Hosting

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

* Add Akka.Management missing AkkaManagementOptions feature
* Add AkkaManagementOptions unit tests
* Add Akka.Discovery.Config.Hosting
* Add Akka.Discovery.Config.Hosting.Tests
  * Settings tests
  * End-to-end test

### CODE REVIEW

File: src/discovery/config/Akka.Discovery.Config.Hosting/Akka.Discovery.Config.Hosting.csproj
Comment (Aaronontheweb) [inline src/discovery/config/Akka.Discovery.Config.Hosting/Akka.Discovery.Config.Hosting.csproj:1]: Let's fold this into the Akka.Management base project - one less NuGet package for users to worry about.

File: src/management/Akka.Management/AkkaHostingExtensions.cs
Comment (Aaronontheweb) [inline src/management/Akka.Management/AkkaHostingExtensions.cs:1]: LGTM

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Collapse the DLLs and then add some docs to the `README.md` on how to use these settings, but otherwise LGTM

**Comment by Arkatufus** [comment]
Done

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**